testing/cli/test_cli_sct_analyze_lesion.py

```diff
@@ -318,6 +318,53 @@ def test_sct_analyze_lesion_with_template(dummy_lesion, tmp_path, tmp_path_qc):
         assert os.path.isfile(tmp_path / f"{fname}{suffix}")
 
 
+# Each tuple represents the starting coordinates (x, y, z) and dimensions (width, height, depth) of a dummy lesion
+@pytest.mark.parametrize("dummy_lesion", [
+    ([(24, 47, 18), (20, 1, 20)])   # large square-like single-slice lesion covering entire SC cross-section --> the lesion is restricted to the cord by sct_analyze_lesion
+], indirect=["dummy_lesion"])
+def test_sct_analyze_lesion_with_template_complete_lesion(dummy_lesion, tmp_path, tmp_path_qc):
+    """
+    Complete lesion spanning the entire spinal cord cross-section.
+    WM and GM PAM50 regions should be fully occupied by the lesion (i.e., 100% coverage).
+    """
+    # prep the template for use with `-f` argument of sct_analyze_lesion
+    sct_register_to_template.main(argv=['-i', sct_test_path('t2', 't2.nii.gz'),
+                                        '-s', sct_test_path('t2', 't2_seg-manual.nii.gz'),
+                                        '-l', sct_test_path('t2', 'labels.nii.gz'),
+                                        '-t', sct_test_path('template'),
+                                        '-ofolder', str(tmp_path),
+                                        '-qc', tmp_path_qc])
+    sct_warp_template.main(argv=['-d', sct_test_path('t2', 't2.nii.gz'),
+                                 '-w', str(tmp_path/'warp_template2anat.nii.gz'),
+                                 '-a', '1',  # TODO: add "small" atlas
```

Comment:
@joshuacwnewton, do you have a sense of how difficult it might be to add the “small” atlas?